src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -512,49 +587,62 @@ exprt string_constraint_generatort::add_axioms_for_characters_in_integer_string(
 
     if(i==0)
     {
-      no_overflow=true_exprt();
       radix_to_power_of_i=from_integer(1, type);
     }
     else
     {
-      const exprt radix_to_power_of_i_minus_one=radix_to_power_of_i;
+      if(i>=overflow_check_index)
+      {
+        /// Keep a record of the previous value for overflow calculation
+        radix_to_power_of_i_minus_one=radix_to_power_of_i;
+      }
       /// Note that power_exprt is probably designed for floating point. Also,
       /// it doesn't work when the exponent isn't a constant, hence why this
       /// loop is indexed by i instead of j. It is faster than
       /// mult_exprt(radix_to_power_of_i, radix).
-      radix_to_power_of_i=power_exprt(radix, i_expr);
-      /// The first time there is overflow we will have that
-      /// radix^i/radix != radix^(i-1)
-      /// However, that condition may hold in future, so we have to be sure to
-      /// propagate the first time this fails to all higher values of i
-      no_overflow=and_exprt(
-        equal_exprt(
-          div_exprt(radix_to_power_of_i, radix), radix_to_power_of_i_minus_one),
-        no_overflow);
+      if(radix_ul==0)
+      {
+        radix_to_power_of_i=power_exprt(radix, i_expr);
```

Comment:
I tried to write some with integer variables but the solver would give incorrect results (looking at bits in the result, it looked like a float).
Maybe it works here because `radix` and `i_expr` are constants but it's better if we can avoid relying on that.